### ISSUE TITLE
@storybook/builder-webpack5 doesn't allow React 18

### ISSUE THREAD

**Issue by DoneDeal0** [storybookjs/storybook#17962]
I can't install `@storybook/builder-webpack5` with npm because of a dependency error:

```js
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: my-app@1.0.0
npm ERR! Found: react@18.0.0
npm ERR! node_modules/react
npm ERR!   react@"^18.0.0" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^16.8.0 || ^17.0.0" from @storybook/builder-webpack5@6.4.22
npm ERR! node_modules/@storybook/builder-webpack5
npm ERR!   dev @storybook/builder-webpack5@"^6.4.21" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
```
It works with yarn install though. 

Could you allow React 18?

EDIT: I see that react 18 is already allowed as a peer dependency in https://github.com/storybookjs/storybook/blob/next/lib/builder-webpack5/package.json . It's really weird.

**Comment by reneaaron** [comment]
I just had the same issue. While installation didn't work with `npm install` it *did* work with `yarn install` 🤷‍♂️

**Comment by brambex** [comment]
same issue here

**Comment by w3bdesign** [comment]
@DoneDeal0 
The next branch is the development branch. 

React 18 support is in development last time I checked.

If you really need support for React 18, either try ```npm install --legacy-peer-deps``` or try ```npx sb upgrade --prerelease```

**Comment by w3bdesign** [comment]
Reference:

https://github.com/storybookjs/storybook/issues/17831

**Comment by TomasKomprs** [comment]
Same issue

**Comment by valentinpalkovic** [comment]
@DoneDeal0 My assumption is, that you are not using the latest prerelease version of storybook (`npx sb upgrade --prerelease`). React 18 is not working with `6.4.x` versions of storybook.

**Comment by usageness** [comment]
I had same issue with same version of storybook.

**Comment by shilman** [comment]
Try upgrading to the latest prerelease:

```
npx sb@next upgrade --prerelease
```

**Comment by nick-bull** [comment]
Specifying a higher version worked for me

```
npm i @storybook/builder-webpack5@6.5.0-beta.4
```

**Comment by nick-bull** [comment]
Specifying a higher version worked for me:
```
npm i -D @storybook/builder-webpack5@6.5.0-beta.4
```

**Comment by ndelangen** [comment]
This has been fixed in storybook 7.0 beta
